packages/react/src/components/TimePicker/ListSpinner.jsx

```diff
@@ -0,0 +1,340 @@
+  onClick: () => {},
+  list: [],
+  containerElement: 'ul',
+};
+
+export function backwardArraySwap(arr) {
+  const newArr = [...arr];
+  const first = newArr.shift();
+  newArr.push(first);
+  return newArr;
+}
+
+export function forwardArraySwap(arr) {
+  const newArr = [...arr];
+  const last = newArr.pop();
+  newArr.unshift(last);
+  return newArr;
+}
+
+/**
+ *
+ * @param {*} arr - the array to opperate on
+ * @param {*} index - the index of the item to move
+ * @returns array - new array with the chosen index as 3 item in array
+ */
+export function moveToSecondIndex(arr, index) {
+  const newArr = [...arr];
+  let removed;
+  if (index > 2) {
+    // 6 - 2
+    removed = newArr.splice(0, index - 2);
+    newArr.push(...removed);
+  }
+  if (index < 2) {
+    const amnt = 2 - index;
+    // 6 - 2
+    removed = newArr.splice(-amnt, amnt);
+    newArr.unshift(...removed);
+  }
+  return newArr;
+}
+
+const ListSpinner = React.forwardRef(
+  (
+    {
+      className,
+      testId,
+      i18n,
+      defaultSelectedId,
+      onChange,
+      onClick,
+      list,
+      containerElement: ContainerElement,
+    },
+    ref
+  ) => {
+    const containerRef = useRef();
+    const contentRef = useRef();
+    const touch = useRef(false);
+    const ticking = useRef(false);
+    const scrollEvent = useRef(false);
+    const scrollPosition = useRef();
+    const timer = useRef();
+
+    const [listItems, setListItems] = useState(list);
+    const [selectedId, setSelectedId] = useState(defaultSelectedId);
+
+    const { previous, next } = useMerged(defaultProps.i18n, i18n);
+
+    useEffect(() => {
+      setSelectedId(defaultSelectedId);
+    }, [defaultSelectedId]);
+
+    useEffect(() => {
+      const index = list.findIndex((i) => i.id === selectedId);
+      const newList = moveToSecondIndex(list, index);
+      setListItems(newList);
+      onChange(selectedId);
+      /* eslint-disable-next-line react-hooks/exhaustive-deps */
+    }, [onChange, selectedId]);
+
+    const observerRef = React.useRef();
+    useEffect(() => {
+      // Create a new observer
+      if (window?.IntersectionObserver) {
+        observerRef.current = new IntersectionObserver(
+          (entries) => {
+            entries.forEach((entry) => {
+              if (entry.isIntersecting && scrollEvent.current) {
+                clearTimeout(timer.current);
+                timer.current = setTimeout(setSelectedId(entry.target.childNodes[0].id), 800);
+              }
+            });
+          },
+          {
+            root: containerRef.current,
+            rootMargin: '-50% 0% -50% 0%',
+            threshold: 0,
+          }
+        );
+      }
+
+      return () => observerRef.current?.disconnect();
+    }, []);
+
+    const listRef = useCallback(
+      (node) => {
+        if (node) {
+          const element = node;
+          containerRef.current = node;
+          element.style.right = `${node.clientWidth - node.offsetWidth}px`;
```

Comment:
hides the scrollbars for each spinner

Reply:
There is a gap here on mobile. Opened an issue [here](https://github.com/carbon-design-system/carbon-addons-iot-react/issues/3508) to track the exploration. From the exploration I have already taken, all css methods are not supported. We will probably have to do through javascript.